app/tests/src/test_chainlit.py

```diff
@@ -56,3 +60,62 @@ def test__get_retrieval_metadata(chunks_with_scores):
         }
         for citations in subsections
     ]
+
+
+def test__get_raw_chat_history():
+    clMessage1 = MagicMock()
+    clMessage2 = MagicMock()
+    clMessage3 = MagicMock()
+    clMessage1.configure_mock(
+        content="CA EDD Web Chat Engine started ",
+        metadata={
+            "engine": "ca-edd-web",
+            "settings": {
+                "llm": "gpt-4o",
+                "retrieval_k": 50.0,
+                "retrieval_k_min_score": -1.0,
+                "chunks_shown_max_num": 8.0,
+                "chunks_shown_min_score": -1.0,
+                "system_prompt": PROMPT,
+            },
+        },
+        author="backend",
+        type="assistant_message",
+    )
+    clMessage2.configure_mock(
+        content="can you tell me about income eligibility?",
+        author="User",
+        type="user_message",
+    )
+    clMessage3.configure_mock(
+        content="<div>To be eligible for unemployment benefits, you need to have earned enough wages during a specific 12-month period called the base period (citation-46). <p>For Disability Insurance, you must have earned at least $300 in wages that were subject to State Disability Insurance (SDI) deductions (citation-123). For Paid Family Leave, you also need to have earned at least $300 with SDI deductions during your base period (citation-129).</p> If you are applying for an overpayment waiver, your average monthly income must be less than or equal to the amounts in the Family Income Level Table (citation-118).</div>",
+        metadata={
+            "system_prompt": PROMPT,
+            "chunks": [factories.ChunkFactory.build()],
+            "subsections": [
+                {
+                    "id": "citation-1",
```

Comment:
The LLM generated response doesn't refer to `(citation-1)` and in fact, refers to citations that aren't in `subsections`, which is not typical or intended. I suggest adjusting the generated response or the `subsections`. Note the response and other `subsections[*].text` don't need to be long to test your new function.

Reply:
removed chunks and subsections as they're not key values we're using